app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,28 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.DiffUtil;
+
+public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
+
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        // Проверка на одинаковость элементов, например, сравнение их хеш-кодов
+        return oldItem.hashCode() == newItem.hashCode();
```

Comment:
Так как оба аргумента являются типом `T`, который реализует интерфейс `ListDiffInterface`, здесь нужно вызвать написанную тобой функцию `theSameAs` для проверки идентичности элементов:
```suggestion
        return oldItem.theSameAs(newItem);
```